### PR TITLE
add create-pbuild-config(cpc) command

### PR DESCRIPTION
**Pull request by adrianschroeter**
which allows to create a working config for pbuild tool based on OBS server side data

The cross build support is not yet done

### CODE REVIEW

File: osc/commandline.py [hunk 1]
```diff
@@ -8342,6 +8342,36 @@ def do_whois(self, subcmd, opts):
             if len(user) == 3:
                 print("%s: \"%s\" <%s>" % (user[0], user[1], user[2]))
 
+    @cmdln.alias('cpc')
+    def do_createpbuildconfig(self, subcmd, opts, *args):
```
Comment (dmach) [inline osc/commandline.py:8346]: Wouldn't be better use name with dashes: `create-pbuild-config` ?
There's no explicit support for that in cmdln.py, but I could add it easily.

File: osc/core.py [hunk 2]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
     f = http_GET(u)
     return f.read()
 
+# This is always replacing a possible exiting config for now
+# we could extend the _pbuild file easily, but what should we do with multiple instances of the _config?
+def create_pbuild_config(apiurl, project, repository, arch, project_dir):
+    # get expanded buildconfig for given project / repository
+    bc = get_buildconfig(apiurl, project, repository)
+    if bc == None:
```
Comment (dmach) [inline osc/core.py:6386]: Do not compare with None using `==`.
Please change to `if not bc:`

File: osc/core.py [hunk 3]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
...
+# cross compile setups are not yet supported
+#    for path in root.findall('hostsystem'):
+#        ET.SubElement(preset, 'hostrepo').text = path.get('url')
+    for path in root.findall('path'):
+        ET.SubElement(preset, 'repo').text = path.get('url')
+
+    ET.SubElement(preset, 'arch').text = arch
+    try:
+        ET.indent(tree, '  ')
+    except:
```
Comment (dmach) [inline osc/core.py:6409]: lgtm checker already complained: `1 for Except block handles 'BaseException'`
Could you specify the exception type?

File: osc/core.py [hunk 4]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
...
+        # old versions do not support it, but the file is just not nicely formated then ...
+        None
```
Comment (dmach) [inline osc/core.py:6411]: Replace with `pass` ?

**Review by dmach — CHANGES_REQUESTED** [review]

File: osc/commandline.py [hunk 5]
```diff
@@ -8342,6 +8342,36 @@ def do_whois(self, subcmd, opts):
...
+
+        Examples:
+            osc cpc REPOSITORY ARCH
+        """
+
+        apiurl = self.get_api_url()
+        if len(args) != 2:
+            raise oscerr.WrongArgs('Wrong number of arguments, require a repository and architecture')
+        repository = args[0]
+        arch = args[1]
```
Comment (dirkmueller) [inline osc/commandline.py:8361]: ```repository, arch = args```

File: osc/core.py [hunk 6]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
     f = http_GET(u)
     return f.read()
 
+# This is always replacing a possible exiting config for now
+# we could extend the _pbuild file easily, but what should we do with multiple instances of the _config?
+def create_pbuild_config(apiurl, project, repository, arch, project_dir):
```
Comment (dirkmueller) [inline osc/core.py:6383]: add "_" as prefix?

File: osc/core.py [hunk 7]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
     f = http_GET(u)
     return f.read()
 
+# This is always replacing a possible exiting config for now
```
Comment (dirkmueller) [inline osc/core.py:6381]: move comment into a ```""""..."""``` section at the beginning of the function

File: osc/core.py [hunk 8]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
...
+# we could extend the _pbuild file easily, but what should we do with multiple instances of the _config?
+def create_pbuild_config(apiurl, project, repository, arch, project_dir):
+    # get expanded buildconfig for given project / repository
+    bc = get_buildconfig(apiurl, project, repository)
+    if bc == None:
+        e.osc_msg = 'failed to get build config for project \'%s\' repository \'%s\'' % (project, repository)
+        raise
+
+    file = open(project_dir + '/_config', 'w')
```
Comment (dirkmueller) [inline osc/core.py:6390]:
``` with open(...) as f:
       f.write(decode_it(bc))
```

avoids the flush and the fd leak

File: osc/core.py [hunk 9]
```diff
@@ -6378,6 +6378,39 @@ def get_buildconfig(apiurl, prj, repository, path=None):
...
+    for path in root.findall('path'):
+        ET.SubElement(preset, 'repo').text = path.get('url')
+
+    ET.SubElement(preset, 'arch').text = arch
+    try:
+        ET.indent(tree, '  ')
+    except:
+        # old versions do not support it, but the file is just not nicely formated then ...
+        None
+    tree.write(project_dir + '/_pbuild', encoding="utf-8", xml_declaration=True)
```
Comment (dirkmueller) [inline osc/core.py:6412]:
```suggestion
    tree.write(os.path.join(project_dir, '_pbuild'), encoding="utf-8", xml_declaration=True)
```

**Comment by adrianschroeter** [comment]
the _ prefix for the method is not working, but if you mind it, we can just move the code into commandline

apart from that I should be done, @dmach please take over or continue to complain :)

File: osc/core.py (hunk 4 above)
Comment (dmach) [reply osc/core.py]: I've removed this code, replacing it with `xmlindent(tree)`

File: osc/commandline.py (hunk 5 above)
Comment (dmach) [reply osc/commandline.py]: no longer needed, args are handled via argparse and that includes throwing an error if they're not specified

**Review by dmach — APPROVED** [review]

**Pull request merged**